sdk/synapse/azure-analytics-synapse-accesscontrol/src/main/java/com/azure/analytics/synapse/accesscontrol/AccessControlAsyncClient.java

```diff
@@ -0,0 +1,497 @@
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.analytics.synapse.accesscontrol;
+
+import com.azure.analytics.synapse.accesscontrol.implementation.AccessControlsImpl;
+import com.azure.analytics.synapse.accesscontrol.implementation.models.AccessControlsGetRoleAssignmentsResponse;
+import com.azure.analytics.synapse.accesscontrol.implementation.models.ErrorContractException;
+import com.azure.analytics.synapse.accesscontrol.implementation.models.RoleAssignmentDetails;
+import com.azure.analytics.synapse.accesscontrol.implementation.models.RoleAssignmentOptions;
+import com.azure.analytics.synapse.accesscontrol.implementation.models.RolesListResponse;
+import com.azure.analytics.synapse.accesscontrol.implementation.models.SynapseRole;
+import com.azure.core.annotation.BodyParam;
+import com.azure.core.annotation.Delete;
+import com.azure.core.annotation.ExpectedResponses;
+import com.azure.core.annotation.Get;
+import com.azure.core.annotation.HeaderParam;
+import com.azure.core.annotation.Host;
+import com.azure.core.annotation.HostParam;
+import com.azure.core.annotation.PathParam;
+import com.azure.core.annotation.Post;
+import com.azure.core.annotation.QueryParam;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceInterface;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.annotation.UnexpectedResponseExceptionType;
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.core.http.rest.PagedResponseBase;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.RestProxy;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import java.util.List;
+import reactor.core.publisher.Mono;
+
+/**
+ * Initializes a new instance of the asynchronous AccessControlClient type.
+ */
+@ServiceClient(builder = AccessControlClientBuilder.class)
+public final class AccessControlAsyncClient {
+    private AccessControlsImpl serviceClient;
+
+    /**
+     * Initializes an instance of AccessControls client.
+     */
+    AccessControlAsyncClient(AccessControlsImpl serviceClient) {
+        this.serviceClient = serviceClient;
+    }
+
+    /**
+     * Create role assignment.
+     * 
+     * @param options Role Assignment request details.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return role Assignment response details.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<SimpleResponse<RoleAssignmentDetails>> createRoleAssignmentWithResponse(RoleAssignmentOptions options) {
```

Comment:
RoleAssignmentDetails is a class in the implementation package, which means it cannot be exposed through public API. You need to move RoleAssignmentDetails out of the implementation package if you intend for it to be used by developers.

Reply:
Can we expose all generated classes?